src/PrestaShopBundle/Routing/Converter/RoutingCacheKeyGenerator.php

```diff
@@ -0,0 +1,132 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Routing\Converter;
+
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+class RoutingCacheKeyGenerator implements CacheKeyGeneratorInterface
+{
+    /**
+     * @var array
+     */
+    private $coreRoutingPaths;
+
+    /**
+     * @var array
+     */
+    private $activeModulesPaths;
+
+    /**
+     * @var string
+     */
+    private $environment;
+
+    /**
+     * RoutingCacheKeyGenerator constructor.
+     *
+     * @param array $coreRoutingPaths
+     * @param array $activeModulesPaths
+     * @param string $environment
+     */
+    public function __construct(
+        array $coreRoutingPaths,
+        array $activeModulesPaths,
+        $environment = 'dev'
+    ) {
+        $this->coreRoutingPaths = $coreRoutingPaths;
+        $this->activeModulesPaths = $activeModulesPaths;
+        $this->environment = $environment;
+    }
+
+    /**
+     * @return array
+     */
+    public function getLastModifications()
+    {
+        $routingFiles = [];
+
+        if (count($this->coreRoutingPaths)) {
+            $finder = new Finder();
+            $finder->files()->in($this->coreRoutingPaths);
+            $finder->name('/\.(yml|yaml)$/');
+            /** @var SplFileInfo $yamlFile */
+            foreach ($finder as $yamlFile) {
+                $routingFiles[$yamlFile->getPathname()] = $yamlFile->getMTime();
+            }
+        }
+
+        foreach ($this->activeModulesPaths as $modulePath) {
+            $routingFile = $modulePath . '/config/routes.yml';
+            if (file_exists($routingFile)) {
+                $routingFiles[$routingFile] = filemtime($routingFile);
+            }
+            $routingFile = $modulePath . '/config/routes.yaml';
+            if (file_exists($routingFile)) {
+                $routingFiles[$routingFile] = filemtime($routingFile);
+            }
+        }
+
+        arsort($routingFiles);
+
+        return $routingFiles;
+    }
+
+    /**
+     * @return array|null
```

Comment:
isn't cleaner to return an empty array instead?

Reply:
I think we will never agree on this kind of behavior :P 
I prefer to return null when no results is found, because I think it holds more signification about the absence of result than an empty array. And I like comparing a variable to null better than using `empty()` But it's matter of preferences of course, I know you don't like multiple return types ^^

Reply:
I side with jolelievre :) function name is `getLatestModification()`. We expect to get one single result. If it cant be found, `null` is an acceptable answer (or an exception throwing). IMO an empty array is irrelevant here (it would be relevant for `getLatestModifications()`).